File: programmers/scala/Level1_1_2.scala
// https://programmers.co.kr/learn/courses/30/lessons/42748?language=scala
// K번쨰 수

import scala.collection.mutable.ListBuffer

object Solution {
    def solution(array: Vector[Int], commands: Vector[Vector[Int]]): Vector[Int] = {
        
        var result_index:ListBuffer[Int] = ListBuffer()
        
        for (cmd <- commands)
          result_index = result_index += (array.slice(cmd(0)-1, cmd(1)).sortWith(_<_)(cmd(2)-1))
        
        return result_index.toVector
        
    }
}

File: programmers/python/Level1_1_2.py
# https://programmers.co.kr/learn/courses/30/lessons/42748?language=scala
# K번째 수
def solution(array, commands):
    answer = []
    
    for command in commands:
        temp_array = array[(command[0]-1):(command[1])]
        temp_array.sort()
        answer.append(temp_array[command[2]-1])
    
    return answer